### PR TITLE
KDEV 1626: Add login hooks properties

### PR DESCRIPTION
**Pull request by mkapostolov**
https://kinvey.atlassian.net/browse/KDEV-1626

### CODE REVIEW

**Review by alexanderfilipov — APPROVED** [review]
LGTM.

**Comment by mkapostolov** [comment]
Added logic to remove username and userId.
Add tests.

File: test/unit/lib/httpReceiver.test.js [hunk 1]
```diff
@@ -437,6 +437,57 @@ describe('http receiver', () => {
     });
   });
 
+  it('should add loginOptinos and remove username and userId for loginHooks', (done) => {
```
Comment (alexanderfilipov) [inline test/unit/lib/httpReceiver.test.js:440]:
```suggestion
  it('should add loginOptions and remove username and userId for loginHooks', (done) => {
```

File: test/unit/lib/httpReceiver.test.js [hunk 2]
```diff
@@ -437,6 +437,57 @@ describe('http receiver', () => {
...
+          should.not.exist(res.body.request.userId);
+          should.not.exist(res.body.request.username);
+          res.body.request.loginOptions.should.be.an.Object();
+          res.statusCode.should.eql(200);
+          done();
+        });
+    });
+  });
+
+  it('should not return loginOptinos and set username and userid for non loginHooks', (done) => {
```
Comment (alexanderfilipov) [inline test/unit/lib/httpReceiver.test.js:466]:
```suggestion
  it('should not return loginOptions and set username and userid for non loginHooks', (done) => {
```

**Review by alexanderfilipov — APPROVED** [review]
LGTM.

**Review by alexanderfilipov — APPROVED** [review]

**Review by alexanderfilipov — APPROVED** [review]

**Pull request merged**